src/topology/running.rs

```diff
@@ -399,20 +416,20 @@ impl RunningTopology {
 
         // Cleanup changed and collect buffers to be reused
         let mut buffers = HashMap::new();
-        for name in &diff.sinks.to_change {
-            if wait_for_sinks.contains(name) {
-                let previous = self.tasks.remove(name).unwrap();
-                debug!(message = "Waiting for sink to shutdown.", %name);
+        for id in &diff.sinks.to_change {
+            if wait_for_sinks.contains(id) {
+                let previous = self.tasks.remove(id).unwrap();
+                debug!(message = "Waiting for sink to shutdown.", name = ?id.name);
```

Comment:
A `Display` implementation for `ComponentId` would also mean none of these have to change.